pkg/stub/stub.go

```diff
@@ -319,7 +318,12 @@ func (stub *stub) Start(ctx context.Context) (retErr error) {
 	if stub.started {
 		return fmt.Errorf("stub already started")
 	}
-	stub.started = true
+	defer func() {
```

Comment:
Why to do it this way ? AFAICT, there is only one exit path where we return a nil error, so it would be much easier to follow that we only set `stub.started` to `true` only on that path, if that happened in a more direct/less convoluted way. Also, all the other deferred functions are cleanups, so they have the opposite logic: they do something on failure, not on success. Moving this to be a direct assignment to the very end would keep that consistently intact.

Reply:
you are right, changed